### ISSUE TITLE
Thread '<unnamed>' panicked at 'State inconsistency: invalid reach attempt cancel: ()', libcore/result.rs:1009

### ISSUE THREAD

**Issue by Genysys** [paritytech/polkadot#21]
OS: Ubuntu 16.0 LTS
Polkadot Version:  parity-polkadot v0.2.16

I attempting to sync with the network, I have come across this error which seems to be spurious:

```
2018-09-16 11:32:56 Syncing, target=#1407541 (8 peers), best: #720271 (9a4e…16ac)
2018-09-16 11:33:01 Syncing, target=#1407542 (8 peers), best: #720326 (9864…7ac0)

====================

stack backtrace:
   0:     0x55efaa020d1c - backtrace::backtrace::trace::h648cb776e96afd21
   1:     0x55efaa01ffd2 - <backtrace::capture::Backtrace as core::default::Default>::default::h82fbb8ebfa0d4c13
   2:     0x55efaa020048 - backtrace::capture::Backtrace::new::h0e4d4489bcf4d061
   3:     0x55efa9904590 - substrate_cli::panic_hook::panic_hook::he795dc17f95aaf04
   4:     0x55efa99042a8 - core::ops::function::Fn::call::hf5571a2e2ac0abe2
   5:     0x55efaa1607c3 - std::panicking::rust_panic_with_hook::hd04e46cfa0b30fca
                        at libstd/panicking.rs:481
   6:     0x55efaa160369 - std::panicking::continue_panic_fmt::ha2ede74723cdc31c
                        at libstd/panicking.rs:391
   7:     0x55efaa160265 - rust_begin_unwind
                        at libstd/panicking.rs:326
   8:     0x55efaa1b9ecb - core::panicking::panic_fmt::hf3e826a5d8e962e1
                        at libcore/panicking.rs:77
   9:     0x55efa9c73855 - core::result::unwrap_failed::hbefe5b1ea074eb92
  10:     0x55efa9c63988 - <libp2p_core::nodes::swarm::Swarm<TTrans, TMuxer, TUserData>>::handle_node_reached::h9177527572f9e620
  11:     0x55efa9c469a8 - <libp2p_core::nodes::swarm::Swarm<TTrans, TMuxer, TUserData> as futures::stream::Stream>::poll::h3d9e903ce9766a3f
  12:     0x55efa9cd4273 - <substrate_network_libp2p::swarm::Swarm<TUserData> as futures::stream::Stream>::poll::h4cc0344f8b44f3b0
  13:     0x55efa9c87dc9 - <substrate_network_libp2p::service_task::Service as futures::stream::Stream>::poll::h423ee0a9c933e716
  14:     0x55efa9c62d23 - <futures::stream::poll_fn::PollFn<F> as futures::stream::Stream>::poll::ha1c09aea94383516
  15:     0x55efa9cbb698 - <futures::stream::for_each::ForEach<S, F, U> as futures::future::Future>::poll::hffdc90c84bf4ed86
  16:     0x55efa9d38efc - <futures::future::select_all::SelectAll<A> as futures::future::Future>::poll::hbdf8f8d43a8f6e81
  17:     0x55efa9db5dee - <futures::future::chain::Chain<A, B, C>>::poll::hfa5930913e069d9b
  18:     0x55efa9d1d74e - <futures::future::map_err::MapErr<A, F> as futures::future::Future>::poll::hf4a898820ab3af90
  19:     0x55efa9c64ac3 - <futures::future::select::Select<A, B> as futures::future::Future>::poll::h1f2390fb29806220
  20:     0x55efa9d7357b - <futures::future::map::Map<A, F> as futures::future::Future>::poll::h31852609c2bb8da7
  21:     0x55efa9d1b317 - <futures::future::map_err::MapErr<A, F> as futures::future::Future>::poll::h3825263f06eb1961
  22:     0x55efa9d2b4ba - futures::task_impl::std::set::h8cd2b7c93c0a0a43
  23:     0x55efa9c49e0c - <std::thread::local::LocalKey<T>>::with::ha2491f8334b5fab2
  24:     0x55efa9cc7d5c - <tokio::executor::current_thread::Entered<'a, P>>::block_on::h54c534cd82b1681e
  25:     0x55efa9c4a0cf - <std::thread::local::LocalKey<T>>::with::hbfffdc1bca2b229e
  26:     0x55efa9c57772 - tokio::runtime::current_thread::runtime::Runtime::block_on::hdc8b4dcb9173d4d0
  27:     0x55efa9d019ba - std::sys_common::backtrace::__rust_begin_short_backtrace::h062c4ca5bf867efd
  28:     0x55efa9c49bf5 - std::panicking::try::do_call::h01a2373f021281c9
  29:     0x55efaa182ce9 - __rust_maybe_catch_panic
                        at libpanic_unwind/lib.rs:102
  30:     0x55efa9d2a9d0 - <F as alloc::boxed::FnBox<A>>::call_box::h1e81c4e7ae73e534
  31:     0x55efaa17437a - <alloc::boxed::Box<(dyn alloc::boxed::FnBox<A, Output$u3d$R$GT$$u20$$u2b$$u20$$u27$a$RP$$GT$$u20$as$u20$core..ops..function..FnOnce$LT$A$GT$$GT$::call_once::hba950b0efee6dead
                        at liballoc/boxed.rs:656
                         - std::sys_common::thread::start_thread::h851ae18c961caf42
                        at libstd/sys_common/thread.rs:24
  32:     0x55efaa160fe5 - std::sys::unix::thread::Thread::new::thread_start::hd5d0edde91f39da6
                        at libstd/sys/unix/thread.rs:90
  33:     0x7f14da0e56b9 - start_thread
  34:     0x7f14d9c0541c - clone
  35:                0x0 - <unknown>

Thread '<unnamed>' panicked at 'State inconsistency: invalid reach attempt cancel: ()', libcore/result.rs:1009
```

Worried as I am unsure if this could lead to slashing of my stake if I become a validator

**Comment by tomaka** [comment]
Has been fixed in substrate, in both the `v0.2` and `master` branches.

**Comment by Genysys** [comment]
Thanks . Curious to the root cause of the problem
